src/myspace/AdditionalInfo.jsx

```diff
@@ -1,21 +1,143 @@
-import React from "react";
+import React, { useState, useEffect, useContext } from "react";
+import { BASE_API } from "../config";
+import { AuthContext } from "../AuthContext";
+import "./MySpace.css";
 
 export default function AdditionalInfo() {
-    return (
-        <div className="container-fluid" id="profile">
-          <div className="top">
-            <h1>
-              This is the Private page for Member's Additional Info page
-            </h1>
-          </div>
-          <div className="middle">
-            <h2>The site is currently under construction...</h2>
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [additionalInfo, setAdditionalInfo] = useState({});
+  const { access_token, user } = useContext(AuthContext);
+
+  const requestAdditionalInfo = {
+    method: "GET",
+    headers: {
+      "Authorization": `Bearer ${access_token}`,
+      "Accept": "application/json",
+      "Content-Type": "application/json",
+    },
+  };
+
+  let data = "";
+  useEffect(() => {
+    fetch(`${BASE_API}/user/additional_info`, requestAdditionalInfo)
+      .then(async response => {
+        data = await response.json();
+        data.message ? setErrorMessage(data.message) : setAdditionalInfo(data);
```

Comment:
error message is already getting set in catch right? would there be a case that on error the control would not go to a catch? Just wondering?

Reply:
@meenakshi-dhanani . For some reason, it seems like only the uncustomised errors get inside catch. where as the customised errors did not get inside catch. you can give it a test this by running the app, login, but skip going to personal details page and go straight to additional info. with error also set inside get, you'll see the error message. but if you comment it out and just relying from the one inside catch, you'll not see the error message. I don't know why this happen, but maybe just the way `fetch` is, or perhaps something in backend need to be change. Any suggestions? cc @ramitsawhney27

Reply:
@mtreacy002 I checked fetch it works that way. Doesn't reject a 400. I'm looking at why and figuring another way. I'll post links on this soon.

Reply:
I checked the documentation - https://javascript.info/fetch. Fetch doesn't reject http status errors. Usually how we handle it is since this same logic will be repeated for every call, we have a wrapper over api calls. You can push this and I can open another PR for that. However even in this, what you'd want to do it, check if the !response.ok and reject the promise with the data. Let catch then be the only place where you setErrorMessage. Once you reject anything, it will go to the catch

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject - Inside the new error you can give the data from response.json

Reply:
got it. made the change. will push now